File: CucumberJVM/Chapter15/15.7-StubDependencyPattern/After/src/test/java/com/wimp/specs/drivers/NotificationsApiDriver.java
package com.wimp.specs.drivers;

import com.wimp.app.models.Notification;
import com.wimp.specs.support.LambdaAction;
import com.wimp.specs.support.RestApiContext;
import com.wimp.specs.support.TestAction;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class NotificationsApiDriver {
    private final RestApiContext restApiContext;

    public NotificationsApiDriver(RestApiContext restApiContext) {
        this.restApiContext = restApiContext;
    }

    public TestAction<List<Notification>> getNotifications(String customerName) {
        return new LambdaAction<>(() -> List.of(restApiContext.getRequest(
            "/api/notifications/" + customerName,
            Notification[].class)));
    }
}

File: Reqnroll/Chapter15/15.4-TestActionPattern/After/WIMP.Specs/Drivers/NotificationsApiDriver.cs
using WIMP.App.Models;
using WIMP.Specs.Support;

namespace WIMP.Specs.Drivers;

public class NotificationsApiDriver(RestApiContext restApiContext)
{
    public TestAction<IReadOnlyCollection<Notification>> GetNotifications(string customerName) =>
        new LambdaAction<IReadOnlyCollection<Notification>>("Get notifications", async () =>
            await restApiContext.GetRequest<Notification[]>(
                $"/api/notifications/{customerName}"));
}
